.github/workflows/g-code-http-tests.yaml

```
@@ -0,0 +1,148 @@
+name: HTTP G-Code Tests
+
+on:
+  # Run on any change to the api directory
+  pull_request:
+    paths:
+      - 'api/**'
+      - 'g-code-testing/**'
+      - 'Makefile'
+    branches:
+      - '*'
+  push:
+    paths:
+      - 'api/**'
```

You probably want to trigger it on this workflow file too
```suggestion
      - 'api/**'
      - '.github/workflows/g-code-http-tests.yaml'
```